sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -163,6 +163,10 @@ case class BucketSpec(
  * @param tracksPartitionsInCatalog whether this table's partition metadata is stored in the
  *                                  catalog. If false, it is inferred automatically based on file
  *                                  structure.
+ * @param schemaFromTableProps Whether the schema field was obtained by parsing a case-sensitive
```

Comment:
Since table props is an implementation detail, consider naming this `schemaPreservesCase`.